Mastodon/Scene/Discovery/ForYou/DiscoveryForYouViewController.swift

```diff
@@ -117,17 +110,20 @@ extension DiscoveryForYouViewController: ProfileCardTableViewCellDelegate {
     func profileCardTableViewCell(
         _ cell: ProfileCardTableViewCell,
         profileCardView: ProfileCardView,
-        relationshipButtonDidPressed button: ProfileRelationshipActionButton
+        relationshipButtonDidPressed button: UIButton
     ) {
         guard let indexPath = tableView.indexPath(for: cell) else { return }
-        guard case let .user(record) = viewModel.diffableDataSource?.itemIdentifier(for: indexPath) else { return }
-        
+        guard case let .account(account, _) = viewModel.diffableDataSource?.itemIdentifier(for: indexPath) else { return }
+
+        cell.profileCardView.setButtonState(.loading)
+
         Task {
-            try await DataSourceFacade.responseToUserFollowAction(
-                dependency: self,
-                user: record
-            )
-        }   // end Task
+            let newRelationship = try await DataSourceFacade.responseToUserFollowAction(dependency: self, user: account)
+
+            await MainActor.run {
+                cell.profileCardView.updateButtonState(with: newRelationship, isMe: false)
```

Comment:
❓  Are you sure there is no need to check for `account.id == authContext...user.id` for `isMe`?

Reply:
Honestly: No idea 😁 

My thoughts were: Why should my instance recommend me to myself? This would make any sense! But now I should think about it. Thank you for pointing this out.